CMakeLists.txt

```
@@ -0,0 +1,127 @@
+cmake_minimum_required(VERSION 3.23)
+project(
+    cppfront
+    LANGUAGES CXX
+    VERSION 0.1.0
+)
+
+option(
+    CPPFRONT_INSTALL_RULES "Enable install rules for cppfront" "${PROJECT_IS_TOP_LEVEL}"
+)
+
+add_executable(cppfront source/cppfront.cpp)
```

```suggestion
add_executable(cppfront_cppfront source/cppfront.cpp)
```

To further recude the chance of possible name collisions, it's good to prefix targets with a project specific name. This prefix can be removed in the install interface with this property:  
https://cmake.org/cmake/help/latest/prop_tgt/EXPORT_NAME.html

---

I agree with this philosophy _except_ for where it conflicts with the actual project name itself. Who is going to import cppfront and then also name a target cppfront?

---

It's probable that there already are CMake buildsystems with a `cppfront` target because cppfront today doesn't provide a CMake buildsystem itself.

---

I personally think this is paranoid and hyper-defensive, but I'm not going to die on this hill :stuck_out_tongue: 

Done.